File: Algorithm/백준/Silver/10866. 덱/덱.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());

        Deque deque = new LinkedList();

        for(int i=0; i<n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();

            if(command.equals("push_front")) {
                deque.addFirst(Integer.parseInt(st.nextToken()));
            } else if(command.equals("push_back")) {
                deque.addLast(Integer.parseInt(st.nextToken()));
            } else if(command.equals("pop_front")) {
                sb.append(deque.isEmpty() ? "-1" : deque.pollFirst());
                sb.append("\n");
            } else if(command.equals("pop_back")) {
                sb.append(deque.isEmpty() ? "-1" : deque.pollLast());
                sb.append("\n");
            } else if (command.equals("size")) {
                sb.append(deque.size());
                sb.append("\n");
            } else if(command.equals("empty")) {
                sb.append(deque.isEmpty() ? "1" : "0");
                sb.append("\n");
            } else if(command.equals("front")) {
                sb.append(deque.isEmpty() ? "-1" : deque.peekFirst());
                sb.append("\n");
            } else if(command.equals("back")) {
                sb.append(deque.isEmpty() ? "-1" : deque.peekLast());
                sb.append("\n");
            }

        } // for
        System.out.println(sb);
    } // main
}

File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <deque>

using namespace std;

int main() {
	deque<int> dq;

	int n; string cmd; int x;
	cin >> n;
	for (int i = 0; i < n; i++) {
		cin >> cmd;
		if (cmd == "push_front") {
			cin >> x;
			dq.push_front(x);
		}
		else if (cmd == "push_back") {
			cin >> x;
			dq.push_back(x);
		}
		else if (cmd == "pop_front") {
			if (dq.empty()) {
				cout << -1 << "\n";
			}
			else {
				cout << dq.front() << "\n";
				dq.pop_front();
			}
		}
		else if (cmd == "pop_back") {
			if (dq.empty()) {
				cout << -1 << "\n";
			}
			else {
				cout << dq.back() << "\n";
				dq.pop_back();
			}
		}
		else if (cmd == "size") {
			cout << dq.size() << "\n";
		}
		else if (cmd == "empty") {
			if (dq.empty()) {
				cout << 1 << "\n";
			}
			else {
				cout << 0 << "\n";
			}
		}
		else if (cmd == "front") {
			if (dq.empty()) {
				cout << -1 << "\n";
			}
			else {
				cout << dq.front() << "\n";
			}
		}		
		else if (cmd == "back") {
			if (dq.empty()) {
				cout << -1 << "\n";
			}
			else {
				cout << dq.back() << "\n";
			}
		}
	}
}
